### PR TITLE
feat: Enable cred setup for workflows created from templates (no-changelog)

### PR DESCRIPTION
**Pull request by tomi**
## Summary

Enable users to open credential setup for workflows that have been created from templates if they skip it.

Next steps (will be their own PRs):
- Add telemetry events
- Add e2e test
- Hide the button when user sets up all the credentials
- Change the feature flag to a new one

https://github.com/n8n-io/n8n/assets/10324676/16c1c42d-e7c7-4beb-8061-c04d4abe55a6

## Related tickets and issues

https://linear.app/n8n/issue/ADO-1637/feature-support-template-credential-setup-for-http-request-nodes-that

## Review / Merge checklist
- [x] PR title and summary are descriptive. **Remember, the title automatically goes into the changelog. Use `(no-changelog)` otherwise.** ([conventions](https://github.com/n8n-io/n8n/blob/master/.github/pull_request_title_conventions.md))
- [ ] [Docs updated](https://github.com/n8n-io/n8n-docs) or follow-up ticket created.
- [ ] Tests included.
   > A bug is not considered fixed, unless a test is added to prevent it from happening again.
   > A feature is not complete without tests.

### CODE REVIEW

File: packages/editor-ui/src/views/SetupWorkflowFromTemplateView/setupTemplate.store.ts [hunk 1]
```diff
@@ -35,131 +25,21 @@ export type RequiredCredentials = {
...
-};
-
 export type AppCredentialCount = {
 	appName: string;
 	count: number;
 };
 
-//#region Getter functions
-
-export const groupNodeCredentialsByKey = (
```
Comment (tomi) [inline packages/editor-ui/src/views/SetupWorkflowFromTemplateView/setupTemplate.store.ts:62]: Everything here has been mainly moved `useCredentialSetupState.ts`

File: packages/editor-ui/src/views/SetupWorkflowFromTemplateView/useCredentialSetupState.ts [hunk 2]
```diff
@@ -0,0 +1,227 @@
...
+	normalizeTemplateNodeCredentials,
+} from '@/utils/templates/templateTransforms';
+import type { INodeCredentialDescription, INodeCredentialsDetails } from 'n8n-workflow';
+import type { NodeTypeProvider } from '@/utils/nodeTypes/nodeTypeTransforms';
+import { getNodeTypeDisplayableCredentials } from '@/utils/nodes/nodeTransforms';
+import sortBy from 'lodash-es/sortBy';
+
+//#region Types
+
+export type NodeCredentials = {
```
Comment (tomi) [inline packages/editor-ui/src/views/SetupWorkflowFromTemplateView/useCredentialSetupState.ts:19]: All of these have been moved (with slight modifications) from `setupTemplate.store.ts`

**Review by mutdmour — COMMENTED** [review]
🚀 Looks good. Just a small question to address

File: packages/editor-ui/src/components/Modal.vue [hunk 3]
```diff
@@ -227,6 +227,7 @@ export default defineComponent({
 
 	.modal-content {
 		overflow: hidden;
+		overflow-y: auto;
```
[the remark this answers is not in the document]
Comment (tomi) [reply packages/editor-ui/src/components/Modal.vue:228]: I checked a few modals but not all. I think it's more problematic if content gets hidden if there's not enough vertical space

**Review by mutdmour — DISMISSED** [review]

**Review by mutdmour — APPROVED** [review]

**Pull request merged**